### ISSUE TITLE
Add offline instances to temporary blocklist

### ISSUE THREAD

**Issue by vertbyqb** [libredirect/libredirect#201]
If an instance is offline, maybe it could optionally be added to an adjustable temporary blocklist so it won't be tried for that amount of time. This would save the time it takes to notice it's offline and redirect to another instance.

**Comment by ManeraKai** [comment]
There's an automatic feature for that called `Auto Redirect of offline instances` in the generals settings.

**Comment by vertbyqb** [comment]
> There's an automatic feature for that called `Auto Redirect of offline instances` in the generals settings.

Yes, but I was suggesting that in addition to that, when an instance is detected as offline it could be added to a temporary blocklist and it would not try to go to that instance in the first place.

For example, say you navigate to a bad site and LibRedirect redirects it to Instance A. It detects instance A is offline and redirects to Instance B, which is online. This is how it currently works. In addition to this, I propose that Instance A could be added to a blocklist with a customizable amount of time, say 1 week, and for that amount of time LibRedirect would not even try to go to Instance A, it would go to Instance B or any other instance. Then after that week is up, LibRedirect would clear the blocklist and be able to try Instance A again.

Checking if an instance is offline and then redirecting if it is wastes time, so forgoing this step would provide a smoother experience on the user's end.

**Comment by ManeraKai** [comment]
> For example, say you navigate to a bad site and LibRedirect redirects it to Instance A. It detects instance A is offline and redirects to Instance B, which is online. This is how it currently works. In addition to this, I propose that Instance A could be added to a blocklist with a customizable amount of time, say 1 week, and for that amount of time LibRedirect would not even try to go to Instance A, it would go to Instance B or any other instance. Then after that week is up, LibRedirect would clear the blocklist and be able to try Instance A again.

This isn't really needed as you could just update the instances list and it will be gone most of the time.

> Checking if an instance is offline and then redirecting if it is wastes time, so forgoing this step would provide a smoother experience on the user's end.

I can lower the number or even remove the 3 seconds waiting screen as it has become more stable. This way you won't even feel that you've been redirected two times.

**Comment by vertbyqb** [comment]
> I can lower the number or even remove the 3 seconds waiting screen as it has become more stable. This way you won't even feel that you've been redirected two times.

This would make the experience much smoother. I might be okay with this as an alternative.

**Comment by ManeraKai** [comment]
I still kept it but in a different way. Because I don't want infinite loops again and I saw a potential of something like that in the code when it detects `browser.webRequest.onErrorOccurred.addListener(` but it just redirects without knowing what the error is and errors differ from browsers. So... I made as if it got redirected 5 times and that will happen fast, it will open that redirection page and counts the timer down. This is a draw :)
